lib/doorkeeper/oauth/authorization_code_request.rb

```diff
@@ -75,9 +75,18 @@ def validate_grant
       end
 
       def validate_redirect_uri
+        # 4.1.3.  Access Token Request
+        #   redirect_uri
+        #     REQUIRED, if the "redirect_uri" parameter was included in the
+        #     authorization request as described in Section 4.1.1, and their
+        #     values MUST be identical.
+        #
+        # @see https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.3
+        return true if redirect_uri.blank? && grant.redirect_uri.blank?
+
         Helpers::URIChecker.valid_for_authorization?(
           redirect_uri,
-          grant.redirect_uri,
+          grant.redirect_uri.presence || client.redirect_uri,
```

Comment:
The purpose of this line of code is to tell Doorkeeper what to do in a situation where the authorization request did not include a `redirect_uri` param but the access token request does include a `redirect_uri` param. As far as I know the OAuth 2.0 specification doesn't say what should happen in that situation.

> redirect_uri
>        #     REQUIRED, if the "redirect_uri" parameter was included in the
>        #     authorization request as described in Section 4.1.1, and their
>        #     values MUST be identical.
https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.3

My options were:
* make the token request fail because the `redirect_uri` param is present
* ignore the presence of the `redirect_uri` param
* validate the `redirect_uri` against the application's pre-registered `redirect_uri` (the same pre-registered `redirect_uri` which Doorkeeper redirected the resource owner to during the authentication request)

I chose the third option.